packages/react/src/tooltip/trigger/TooltipTrigger.tsx

```diff
@@ -46,14 +46,18 @@ namespace TooltipTrigger {
     open: boolean;
   }
 
-  export interface Props extends BaseUIComponentProps<any, State> {}
+  export interface Props extends BaseUIComponentProps<'div', State> {}
```

Comment:
Do we expect here people to always need to cast if they use different element?

Reply:
The `render` prop's `props` parameter is actually loosely typed in order to allow you to compose different tags. 

We need to ensure we document when a tag can't be replaced if we aren't polyfilling native behavior (e.g. if it relies in a specific tag's native props in order to work, like `<button>` being replaced with a `<span>`, but internally is not using `useButton()` to keep it accessible).

We could actually remove the first `tag` type since it serves no purpose in this case

Reply:
Let's see if this will be enough, if we see more issues around this we can revisit it.